app/models/game.rb

```diff
@@ -44,6 +44,18 @@ def next_player(player)
      
   end
 
+  def player_checkmate?(color)
+     game = self
+     king = game.pieces.where(type: "King", color: color)[0]
+     
+     return king.is_in_checkmate?
+  end
+
+  def player_in_check?(color)
+     game = self
+     king = game.pieces.find_by(type: "King", color: color)
+     return true if king.is_in_check?
```

Comment:
Instead of `return true if king.is_in_check?` consider `return king.is_in_check?`; They are almost the same.